codegen/src/api/storage.rs

```diff
@@ -173,6 +170,7 @@ fn primitive_type_alias(type_path: &TypePath) -> TokenStream {
     quote!(#type_path)
 }
 
+#[cfg(feature = "fetch_metadata")]
```

Comment:
I can't quite see why this is needed offhand; we don't fetch any metadata below?

Reply:
Oh, it is just used here, because the `frame_metadata` crate that is used in the tests, is only a dependency if the "fetch_metadata" feature is enabled. Otherwise we do not need it as a dependency.

Reply:
Just make `frame-metadata` a non-optional dev-dependency and remove the added feature should work?